### PR TITLE
Remove concurrency dependency in restfulWSClient 3.1 feature

### PR DESCRIPTION
**Pull request by jhanders34**
#build

### CODE REVIEW

**Comment by jhanders34** [comment]
#build

**Comment by jhanders34** [comment]
#libby #build
#spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.defaultexceptionmapper_fat,io.openliberty.ws.jaxrs.2.1.jaxws.2.2_fat,io.openliberty.restfulWS.3.0.client_fat

**Comment by jhanders34** [comment]
#libby #build
#spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.defaultexceptionmapper_fat,io.openliberty.ws.jaxrs.2.1.jaxws.2.2_fat,io.openliberty.restfulWS.3.0.client_fat

File: dev/io.openliberty.org.jboss.resteasy.common.ee10/src/io/openliberty/org/jboss/resteasy/common/client/LibertyResteasyClientBuilderImpl.java [hunk 1]
```diff
@@ -81,6 +80,18 @@ public ResteasyClient build() {
            }
         }
 
+        if (asyncExecutor == null) {
+            cleanupExecutor = false;
+            OsgiFacade facade = OsgiFacade.instance().orElse(null);
+            if (facade != null) {
+                Object serviceRef = facade.getServiceRef(ExecutorService.class).orElse(null);
```
Comment (njr-11) [inline dev/io.openliberty.org.jboss.resteasy.common.ee10/src/io/openliberty/org/jboss/resteasy/common/client/LibertyResteasyClientBuilderImpl.java]: Is this way of obtaining the Liberty global executor sufficiently non-brittle?  Code elsewhere includes a filter, such as `(component.name=com.ibm.ws.threading)` to ensure the correct one is obtained.

**Review by njr-11 — APPROVED** [review]

File: dev/io.openliberty.org.jboss.resteasy.common.ee10/src/io/openliberty/org/jboss/resteasy/common/client/LibertyResteasyClientBuilderImpl.java (hunk 1 above)
Comment (jhanders34) [reply dev/io.openliberty.org.jboss.resteasy.common.ee10/src/io/openliberty/org/jboss/resteasy/common/client/LibertyResteasyClientBuilderImpl.java]: I tried to find an example of what you are referring to.  I couldn't find one.  Can you point me to an example?

Comment (njr-11) [reply dev/io.openliberty.org.jboss.resteasy.common.ee10/src/io/openliberty/org/jboss/resteasy/common/client/LibertyResteasyClientBuilderImpl.java]: Sure, here are some examples,
https://github.com/OpenLiberty/open-liberty/blob/adec923541de1b7cfa32414304f62f867feea113/dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/services/impl/JPAPersistenceManagerImpl.java#L194
https://github.com/OpenLiberty/open-liberty/blob/adec923541de1b7cfa32414304f62f867feea113/dev/com.ibm.ws.concurrent.persistent/src/com/ibm/ws/concurrent/persistent/internal/PersistentExecutorImpl.java#L1839
https://github.com/OpenLiberty/open-liberty/blob/adec923541de1b7cfa32414304f62f867feea113/dev/com.ibm.ws.threading/src/com/ibm/ws/threading/CompletionStageFactory.java#L47
https://github.com/OpenLiberty/open-liberty/blob/adec923541de1b7cfa32414304f62f867feea113/dev/com.ibm.ws.threading/src/com/ibm/ws/threading/PolicyExecutorProvider.java#L47

But it looks like there are even more examples to the contrary, so it looks like no one worries about this.

**Comment by jhanders34** [comment]
#libby #build
#spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.defaultexceptionmapper_fat,io.openliberty.ws.jaxrs.2.1.jaxws.2.2_fat,io.openliberty.restfulWS.3.0.client_fat

**Comment by jhanders34** [comment]
#libby #build
#spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.defaultexceptionmapper_fat,io.openliberty.ws.jaxrs.2.1.jaxws.2.2_fat,io.openliberty.restfulWS.3.0.client_fat

File: dev/io.openliberty.org.jboss.resteasy.common.ee10/src/io/openliberty/org/jboss/resteasy/common/client/LibertyResteasyClientBuilderImpl.java (hunk 1 above)
Comment (jhanders34) [reply dev/io.openliberty.org.jboss.resteasy.common.ee10/src/io/openliberty/org/jboss/resteasy/common/client/LibertyResteasyClientBuilderImpl.java]: After doing some testing I ended up doing a different way to resolve this.  As such I was able to do what you were asking for.  Can you take another look at the changes?

**Comment by jhanders34** [comment]
#libby #build
#spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.defaultexceptionmapper_fat,io.openliberty.ws.jaxrs.2.1.jaxws.2.2_fat,io.openliberty.restfulWS.3.0.client_fat

**Review by WhiteCat22 — APPROVED** [review]

**Comment by jhanders34** [comment]
#libby #build
#spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.defaultexceptionmapper_fat,io.openliberty.ws.jaxrs.2.1.jaxws.2.2_fat,io.openliberty.restfulWS.3.0.client_fat

**Pull request merged**